src/java.net.http/share/classes/jdk/internal/net/http/Http2Connection.java

```diff
@@ -131,7 +131,7 @@ class Http2Connection  {
 
     private static final int MAX_CLIENT_STREAM_ID = Integer.MAX_VALUE; // 2147483647
     private static final int MAX_SERVER_STREAM_ID = Integer.MAX_VALUE - 1; // 2147483646
-    private IdleConnectionTimeoutEvent idleConnectionTimeoutEvent;  // may be null
+    private volatile IdleConnectionTimeoutEvent idleConnectionTimeoutEvent;  // may be null
```

Comment:
What was the reason for this change? `idleConnectionTimerEvent` is always protected by `stateLock`.

Reply:
Hello Daniel, you are right. I missed cleaning up that part from my early experiments. I'll fix that and do the necessary changes shortly.